site/student-lectures/AIS/ais.rst

```diff
@@ -0,0 +1,163 @@
+   immune system. The algorithms are typically modeled after the immune
+   system's characteristics of learning and memory for use in
+   problem-solving.
+
+   AIS emerged in the mid-1980s with articles authored by Farmer,
+   Packard and Perelson (1986) and Bersini and Varela (1990) on immune
+   networks. However, it was only in the mid-1990s that AIS became a
+   field in its own right.
+
+   Forrest et al. (on negative selection) and Kephart et al.[2]
+   published their first papers on AIS in 1994, and Dasgupta conducted
+   extensive studies on Negative Selection Algorithms.
+
+.. image:: \image001.png
+   :alt: Immune system response
+   :align: center
+
+This is how are immune system response
+
+Now, we will learn about some terms :
+
+.. image:: \image002.png
+   :alt: Comparison between IS and AIS
+   :align: center
+
+**Framework of AIS**
+----------------------
+a.Recognition via regions of complementarity and Shape space(S)
+
+b.Cross reactivity
+
+.. note:: *How do we represent it mathematically?*
+
+   Set of coordinates: m = (m1, m2, m3,….ml),m
+
+   Ab = (Ab1, Ab2,….Abl), Ag = (Ag1, Ag2, …Agl)
+
+
+Types of shape space:
+
+    1.Hamming
+
+    2.Euclidean
+
+    3.Manhattan
+
+.. image:: \image003.png
+   :alt: distances
+   :align: center
+
+
+**Techniques which are inspired by the immune system**
+---------------------------------------------------------
+.. image:: \image005.jpg
+   :alt: Algorithms in AIS
+   :align: center
+
+**1.Clonal selection algorithm:** Clonal selection algorithms are
+   most commonly applied to optimization and pattern recognition.
+
+**2.Negative selection algorithm:** This class of algorithms are
+   typically used for classification and pattern recognition problem
+   domains where the problem space is modeled in the complement of
+   available knowledge. For example, in the case of an anomaly detection
+   domain the algorithm prepares a set of exemplar pattern detectors
+   trained on normal (non-anomalous) patterns that model and detect
+   unseen or anomalous patterns.
+
+**3.Immune network algorithm:** This class of algorithms focus on
+   the network graph structures involved where antibodies (or antibody
+   producing cells) represent the nodes and the training algorithm
+   involves growing or pruning edges between the nodes based on affinity
+   (similarity in the problems representation space). Immune network
+   algorithms have been used in clustering, data visualization, control,
+   and optimization domains, and share properties with artificial neural
+   networks.
+
+**4.Dendritic cell algorithm:** The DCA is abstracted and
+   implemented through a process of examining and modeling various
+   aspects of DC function, from the molecular networks present within
+   the cell to the behaviour exhibited by a population of cells as a
+   whole. Within the DCA information is granulated at different layers,
+   achieved through multi-scale processing.
```

Comment:
This too is taken directly from wikipedia 
https://en.wikipedia.org/wiki/Artificial_immune_system

Reply:
I have changed it in my words